test/docker/kafka/component_integration_test.go

```diff
@@ -0,0 +1,261 @@
+// +build integration
+
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron"
+	"github.com/beatlabs/patron/component/async"
+	"github.com/beatlabs/patron/component/async/kafka"
+	"github.com/beatlabs/patron/component/async/kafka/group"
+	"github.com/stretchr/testify/suite"
+)
+
+type KafkaComponentTestSuite struct {
+	suite.Suite
+}
+
+func TestKafkaComponentTestSuite(t *testing.T) {
+	suite.Run(t, new(KafkaComponentTestSuite))
+}
+
+func (s *KafkaComponentTestSuite) TestKafkaComponent_Success() {
+	// Timeout the test if it takes too long
+	timeoutTimer := testTimeout(s, 15*time.Second)
```

Comment:
I used this to add a timeout to these tests (since it is the only way I found to add timeout per test), if you don't like it I can remove it.

Reply:
you can actually control the timeout via the go test argument from outside. no need to have this internally. I would say we should delete it.